core/sequencer/server.go

```diff
@@ -139,27 +139,38 @@ func NewServer(
 // b) apply the batch to the map
 // c) publish existing map roots to a log of SignedMapRoots.
 func (s *Server) RunBatch(ctx context.Context, in *spb.RunBatchRequest) (*empty.Empty, error) {
-	outstandingRevs, err := s.DefineRevisions(ctx, in.DirectoryId, in.MinBatch, in.MaxBatch)
+	defResp, err := s.loopback.DefineRevisions(ctx, &spb.DefineRevisionsRequest{
+		DirectoryId: in.DirectoryId,
+		MinBatch:    in.MinBatch,
+		MaxBatch:    in.MaxBatch})
 	if err != nil {
 		return nil, err
 	}
 
-	for _, rev := range outstandingRevs {
-		if _, err := s.loopback.CreateRevision(ctx, &spb.CreateRevisionRequest{
+	for _, rev := range defResp.OutstandingRevisions {
+		if _, err := s.loopback.ApplyRevision(ctx, &spb.ApplyRevisionRequest{
 			DirectoryId: in.DirectoryId,
 			Revision:    rev,
 		}); err != nil {
-			return nil, err
+			glog.Errorf("ApplyRevision(dir: %v, rev: %v): %v", in.DirectoryId, rev, err)
```

Comment:
I notice you log and continue to publish the applied revisions, rather than retrying failed ApplyRevision calls. That seems correct to me, but it's somewhat at odds with the comment in the API doc:

> If this method fails, it must be retried with the same arguments.

Should we remove that line from the API?

Reply:
The API line is correct. 

`RunBatch` gets called on a periodic timer.  The next call to `DefineRevisions` will contain the failed revision since it has not yet been applied, and we will try it again.